itou/users/admin_forms.py

```diff
@@ -58,3 +61,57 @@ def clean(self):
             except AddressLookupError:
                 # Nothing to do: re-raised and already logged as error
                 pass
+
+
+FakeField = namedtuple("FakeField", ("name",))
+
+
+class FakeRelForToUserRawIdWidget:
+    model = User
+    limit_choices_to = {
+        "kind": UserKind.JOB_SEEKER,
+    }
+
+    def get_related_field(self):
+        # This must return something that has the name of an existing field
+        return FakeField("id")
+
+
+class ToUserRawIdWidget(widgets.ForeignKeyRawIdWidget):
+    def __init__(self, admin_site, attrs=None, using=None):
+        super().__init__(FakeRelForToUserRawIdWidget(), admin_site, attrs, using)
+
+
+class SelectTargetUserForm(forms.Form):
+    # Needed to avoid RemovedInDjango50Warning
+    template_name = forms.Form.template_name_div
+
+    to_user = forms.ModelChoiceField(
+        User.objects.filter(kind=UserKind.JOB_SEEKER), required=True, label="Choisissez l'utilisateur cible"
```

Comment:
On peut définir le queryset dans `__init__`, de sorte à exclure `from_user` du queryset. Ça évite un cas d’erreur et évite d’avoir à se poser la question en cas de doublon de nom.
```suggestion
        User.objects.filter(kind=UserKind.JOB_SEEKER).exclude(pk=from_user.pk), required=True, label="Choisissez l'utilisateur cible"
```

Reply:
Vu que j'utilise un `ForeignKeyRawIdWidget` il faut faire remonter l'info jusqu'au FakeRelForToUserRawIdWidget et j'avais la flemme :confounded:

Reply:
Et en retestant rapidement, j'ai l'impression qu'on ne peut pas avoir de condition négative dans `limit_choices_to`...

Reply:
Effectivement. On pourrait rajouter un lookup custom `__ne` (untested), mais le jeu n’en vaut pas la chandelle. :+1: